src/mrinufft/trajectories/maths.py

```diff
@@ -210,6 +210,55 @@ def Ra(vector, theta):
 #############
 
 
+def is_from_fibonacci_sequence(n):
+    """Check if an integer belongs to the Fibonacci sequence.
+
+    An integer belongs to the Fibonacci sequence if either
+    :math:`5*n²+4` or :math:`5*n²-4` is a perfect square.
+
+    Parameters
+    ----------
+    n : int
+        Integer to check.
+
+    Returns
+    -------
+    bool
+        Whether or not ``n`` belongs to the Fibonacci sequence.
+    """
+
+    def _is_perfect_square(n):
+        r = int(np.sqrt(n))
+        return r * r == n
+
+    return _is_perfect_square(5 * n**2 + 4) or _is_perfect_square(5 * n**2 - 4)
+
+
+def get_closest_fibonacci_number(n):
+    """Provide the closest Fibonacci number.
+
+    Parameters
+    ----------
+    n : int
+        Integer to match.
+
+    Returns
+    -------
+    int
+        Closest number from the Fibonacci sequence.
+    """
+    # Generate Fibonacci number up to n
```

Comment:
I think you can do better by using F_n ≈ int(phi^n/sqrt(5)) and thus `floor(log_phi(x))` should give you the closest index, from which you can compute F_n. 

This would probably be more performant for large n, but I don't think we care for that here.

Reply:
The check between upper and lower Fibonacci number still seems to be necessary, but getting them is now a constant time computation